nextcord/application_command.py

```diff
@@ -1240,76 +1262,180 @@ def __init__(
+            found_type = MISSING
+            found_choices = []
+            for lit in typing.get_args(parameter.annotation):
+                lit = unpack_annotated(lit, list(self.option_types.keys()))
+                lit_type = type(lit)
+                if lit_type in (int, str, float, type(None)):
+                    if lit is None:
+                        # If None is included, they want it to be optional. But we don't want None added to the choices.
+                        annotation_required = False
+                    else:
+                        if found_type is MISSING:
+                            # If we haven't set the type of the annotation, set it.
+                            found_type = self.get_type(lit_type)
+                        elif self.get_type(lit_type) is not found_type:
+                            raise ValueError(
+                                f"{self.error_name} | Literal {lit} is incompatible with {found_type}"
+                            )
+                        found_choices.append(lit)
+
+                else:
+                    raise ValueError(
+                        f"{self.error_name} Invalid type for choices: {type(lit)}: {lit}"
+                    )
+
+            if found_type is MISSING:
+                # ??? Why are they using Literal???
+                found_type = ApplicationCommandOptionType.string
+
+            annotation_type = found_type
+            annotation_choices = found_choices
+        elif typehint_origin in (Union, Optional, None):
+            # If the typehint base is Union, Optional, or not any grouping...
+            found_type = MISSING
+            found_channel_types = []
+
+            if typehint_origin is None:
+                unpacked_annotations: List[type] = [
+                    parameter.annotation,
+                ]
+                literals: List[Annotated[OptionConverter, object]] = []
+            else:
+                unpacked_annotations, literals = unpack_annotation(
+                    parameter.annotation, list(self.option_types.keys())
+                )
+            # Make sure that all literals are only OptionConverters and nothing else.
+            for lit in literals:
+                if not isinstance(lit, OptionConverter) or lit is not None:
+                    raise ValueError(
+                        f"{self.error_name} You cannot use non-OptionConverter literals when the base annotation is "
+                        f"not Literal."
+                    )
+
+            # Pyright gets upset at appending these lists together, but not upset when .extend is used?
+            # grouped_annotations: List[Union[type, Annotated[object, OptionConverter]]] = unpacked_annotations + \
+            #                                                                              literals
+            grouped_annotations: List[
+                Union[type, Annotated[Optional[OptionConverter], object], Type[None]]
```

Comment:
Why is `Type[None]` used here? Is there a specific reason?

Reply:
Can't remember why exactly right now, but I'm pretty sure it's because the type of None isn't the same as None at all.

Reply:
Ah okay